src/control/ToolHandler.cpp

```diff
@@ -448,13 +448,7 @@ void ToolHandler::loadSettings() {
 
 void ToolHandler::pointCurrentToolToButtonTool() { this->currentTool = this->buttonSelectedTool; }
 
-void ToolHandler::pointCurrentToolToToolbarTool() {
-    this->currentTool = this->toolbarSelectedTool;
-
-    this->listener->toolColorChanged(false);
-    this->listener->toolSizeChanged();
-    this->fireToolChanged();
```

Comment:
I double checked and the `toolColorChanged` and `toolSizeChanged` are superfluous as they are actually called withing `fireToolChanged`.
`fireToolChanged`, however, is important for the toolbar to always show the most current state so this should not be removed.

I also played arround with it and I think following change might help quite a lot performance wise:

```cpp
void ToolHandler::pointCurrentToolToButtonTool(Button button) {
    Tool* tool = getButtonTool(button);
    if (this->activeTool == tool)
        return;
    this->activeTool = tool;

    this->fireToolChanged();
}

void ToolHandler::pointCurrentToolToToolbarTool() {
    if (this->activeTool == this->toolbarSelectedTool)
        return;
    this->activeTool = this->toolbarSelectedTool;

    this->fireToolChanged();
}
```

If you don't mind I would include this change in my current ongoing PR (#2339).
If you could check if my change improves the situation in any way that would be great :nerd_face: